### PR TITLE
Pin typing-extensions 4.2.0 to python>=3.7

### PR DESCRIPTION
**Pull request by synapticarbors**
Deals with an issue raised in/caused by https://github.com/conda-forge/typing_extensions-feedstock/pull/30

I'm not 100% sure if I composed this correctly, so feedback is very much welcome.

### CODE REVIEW

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -1421,6 +1421,14 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
         # conda-libmamba-solver 22.3.1 and prior are not compatible
         if record_name == "conda-libmamba-solver" and record.get("timestamp", 0) <= 1650455037727:
             _replace_pin("libmambapy >=0.22", "libmambapy 0.22.*", record["depends"], record)
+
+        # typing-extensions 4.2.0 requires python >=3.7. Build 0 incorrectly specified >=3.6. Fixed in
+        # https://github.com/conda-forge/typing_extensions-feedstock/pull/30
+        if record_name in ["typing-extensions", "typing_extensions"]:
+            if record["version"] == "4.2.0" and record["build"].endswith("_0"):
+                i = record['depends'].index('python >=3.6')
+                record['depends'][i] = 'python >=3.7'
```
Comment (chrisburr) [inline recipe/gen_patch_json.py:1430]:
```suggestion
                _replace_pin("python >=3.6", "python >=3.7", deps, record)
```

File: recipe/gen_patch_json.py [hunk 2]
```diff
@@ -1421,6 +1421,14 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
         # conda-libmamba-solver 22.3.1 and prior are not compatible
         if record_name == "conda-libmamba-solver" and record.get("timestamp", 0) <= 1650455037727:
             _replace_pin("libmambapy >=0.22", "libmambapy 0.22.*", record["depends"], record)
+
+        # typing-extensions 4.2.0 requires python >=3.7. Build 0 incorrectly specified >=3.6. Fixed in
+        # https://github.com/conda-forge/typing_extensions-feedstock/pull/30
+        if record_name in ["typing-extensions", "typing_extensions"]:
```
Comment (chrisburr) [inline recipe/gen_patch_json.py:1427]: `typing-extensions` doesn't depend on Python:
```suggestion
        if record_name == "typing_extensions":
```

**Comment by synapticarbors** [comment]
@chrisburr -- Thanks for the review. I wasn't aware of the `_replace_pin`, so that's good to know. Is it just a cleaner version of what I wrote originally (based on other fixes I saw), or does it do something different?

**Comment by chrisburr** [comment]
> Is it just a cleaner version of what I wrote originally (based on other fixes I saw), or does it do something different?

Pretty much (though it allows the pin to be missing so technically either of my suggestions would be have fixed it):

https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/e0582db8619d5a4515311317ff186d4eecb6e04b/recipe/gen_patch_json.py#L1484-L1490

The diff looks good so LGTM 👍 

```
+ python show_diff.py
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
noarch::typing_extensions-4.2.0-pyha770c72_0.tar.bz2
-    "python >=3.6"
+    "python >=3.7"
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata.json.bz2
```

**Pull request merged**